motoko/icrc2-swap/README.md

```diff
@@ -0,0 +1,311 @@
+export ALICE=$(dfx identity get-principal --identity alice)
+
+dfx identity new bob
+export BOB=$(dfx identity get-principal --identity bob)
+```
+
+## Step 2: Deploy two tokens
+
+Deploy Token A:
+
+```bash
+dfx deploy --network local token_a --argument '
+  (variant {
+    Init = record {
+      token_name = "Token A";
+      token_symbol = "A";
+      minting_account = record {
+        owner = principal "'${OWNER}'";
+      };
+      initial_balances = vec {
+        record {
+          record {
+            owner = principal "'${ALICE}'";
+          };
+          100_000_000_000;
+        };
+      };
+      metadata = vec {};
+      transfer_fee = 10_000;
+      archive_options = record {
+        trigger_threshold = 2000;
+        num_blocks_to_archive = 1000;
+        controller_id = principal "'${OWNER}'";
+      };
+      feature_flags = opt record {
+        icrc2 = true;
+      };
+    }
+  })
+'
+```
+
+Deploy Token B:
+
+```bash
+dfx deploy --network local token_b --argument '
+  (variant {
+    Init = record {
+      token_name = "Token B";
+      token_symbol = "B";
+      minting_account = record {
+        owner = principal "'${OWNER}'";
+      };
+      initial_balances = vec {
+        record {
+          record {
+            owner = principal "'${BOB}'";
+          };
+          100_000_000_000;
+        };
+      };
+      metadata = vec {};
+      transfer_fee = 10_000;
+      archive_options = record {
+        trigger_threshold = 2000;
+        num_blocks_to_archive = 1000;
+        controller_id = principal "'${OWNER}'";
+      };
+      feature_flags = opt record {
+        icrc2 = true;
+      };
+    }
+  })
+'
+```
+## Step 3: Deploy the swap canister
+
+The swap canister accepts deposits, and performs the swap.
+
+```bash
+export TOKEN_A=$(dfx canister id --network local token_a)
+export TOKEN_B=$(dfx canister id --network local token_b)
+
+dfx deploy --network local swap --argument '
+  record {
+    token_a = (principal "'${TOKEN_A}'");
+    token_b = (principal "'${TOKEN_B}'");
+  }
+'
+
+export SWAP=$(dfx canister id --network local swap)
+```
+
+## Step 4: Approve & deposit tokens
+
+Before we can swap the tokens, they must be transferred to the swap canister.
+With ICRC-2, this is a two-step process. First we approve the transfer:
+
+```bash
+# Approve Bob to deposit 1.00000000 of Token B, and 0.0001 extra for the
+# transfer fee
+dfx canister --network local call --identity alice token_a icrc2_approve '
+  record {
+    amount = 100_010_000;
+    spender = record {
+      owner = principal "'${SWAP}'";
+    };
+  }
+'
+
+# Approve Bob to deposit 1.00000000 of Token B, and 0.0001 extra for the
+# transfer fee
+dfx canister --network local call --identity bob token_b icrc2_approve '
+  record {
+    amount = 100_010_000;
+    spender = record {
+      owner = principal "'${SWAP}'";
+    };
+  }
+'
+```
+
+Then we can perform the deposit to transfer the tokens from our wallet to the swap canister:
```

Comment:
Do you think we could reword this? IMO this sounds like _we_ do the transfer when it is the `swap` canister that performs the transfer. While we two understand this easily, I think it will confuse people that are new to the whole ICRC-2 thing

Reply:
Good point. I tried rewriting this. What do you think?